cmd/object-api-utils.go

```diff
@@ -64,12 +69,13 @@ func IsValidBucketName(bucket string) bool {
 	if len(bucket) < 3 || len(bucket) > 63 {
 		return false
 	}
-	if bucket[0] == '.' || bucket[len(bucket)-1] == '.' {
-		return false
+	labels := strings.Split(bucket, ".")
+	for _, label := range labels {
+		if !validLabel.MatchString(label) {
+			return false
+		}
 	}
-	return (validBucket.MatchString(bucket) &&
-		!isIPAddress.MatchString(bucket) &&
-		!strings.Contains(bucket, ".."))
```

Comment:
We should move this check above the regex check to get some fair optimization - regex usage is memory heavy in Go - can be seen for all concurrent workloads.